### ISSUE TITLE
no error when passing value instead of array of values

### ISSUE THREAD

**Issue by dvic** [99designs/gqlgen#233]
### Expected Behaviour
Given input type `MyInput { items [String!] }`, when I use `{items: "foo"}` I should get an error as I should pass an array of string and not a string.

### Actual Behavior
No errors, but resolver is called with items set to an empty slice.

I guess this should be handled in the parser? Is this properly handled in the new parser https://github.com/vektah/gqlparser? 

### Minimal graphql.schema and models to reproduce
```
schema {
    query: Query
}

type Query {
    getItems(input: InputSlice): [String!]
}

input InputSlice {
	items: [String!]
}
```

**Comment by vektah** [comment]
Yeah, I honestly think this is one of the few missteps in the spec, but I'll make sure this is handled consistently in the next branch.

http://facebook.github.io/graphql/June2018/#sec-Type-System.List

> If the value passed as an input to a list type is not a list and not the null value, then the result of input coercion is a list of size one, where the single item value is the result of input coercion for the list’s item type on the provided value (note this may apply recursively for nested lists).

So in your example passing foo should be valid, and should be getting coerced into a list with one item `["foo"]`. The validator is allowing it in, but nothing in gqlgen is coercing it into a list.

**Comment by vektah** [comment]
Fixed in 0.4.0
